### PR TITLE
[Merged by Bors] - feat: port Data.Set.Intervals.Monotone

### PR DESCRIPTION
**Pull request by winston-h-zhang**
Surprising tricky for a beginner O_O -- had to figure out that `rename_i` was a thing.

### CODE REVIEW

File: Mathlib/Data/Set/Intervals/Monotone.lean [hunk 1]
```diff
@@ -0,0 +1,278 @@
...
+variable {α β : Type _} [SemilatticeSup α] [LinearOrder β] {f g : α → β} {a b : β}
+
+theorem unionᵢ_Ioo_of_mono_of_isGLB_of_isLUB (hf : Antitone f) (hg : Monotone g)
+    (ha : IsGLB (range f) a) (hb : IsLUB (range g) b) : (⋃ x, Ioo (f x) (g x)) = Ioo a b :=
+  calc
+    (⋃ x, Ioo (f x) (g x)) = (⋃ x, Ioi (f x)) ∩ ⋃ x, Iio (g x) :=
+      unionᵢ_inter_of_monotone hf.Ioi hg.Iio
+    _ = Ioi a ∩ Iio b := congr_arg₂ (· ∩ ·) ha.unionᵢ_Ioi_eq hb.unionᵢ_Iio_eq
+
+#align Union_Ioo_of_mono_of_is_glb_of_is_lub unionᵢ_Ioo_of_mono_of_isGLB_of_isLUB
```
Comment (Ruben-VandeVelde) [inline Mathlib/Data/Set/Intervals/Monotone.lean:197]:
```suggestion
#align Union_Ioo_of_mono_of_is_glb_of_is_lub unionᵢ_Ioo_of_mono_of_isGLB_of_isLUB
```

**Review by Ruben-VandeVelde — APPROVED** [review]

File: Mathlib/Data/Set/Intervals/Monotone.lean [hunk 2]
```diff
@@ -0,0 +1,277 @@
...
+
+protected theorem AntitoneOn.Ioo (hf : AntitoneOn f s) (hg : MonotoneOn g s) :
+    MonotoneOn (fun x => Ioo (f x) (g x)) s :=
+  hf.Ioi.inter hg.Iio
+#align antitone_on.Ioo AntitoneOn.Ioo
+
+end Ixx
+
+section Union
+
```
Comment (dupuisf) [inline Mathlib/Data/Set/Intervals/Monotone.lean:187]:
```suggestion
section unionᵢ

```

File: Mathlib/Data/Set/Intervals/Monotone.lean [hunk 3]
```diff
@@ -0,0 +1,277 @@
...
+theorem unionᵢ_Ioo_of_mono_of_isGLB_of_isLUB (hf : Antitone f) (hg : Monotone g)
+    (ha : IsGLB (range f) a) (hb : IsLUB (range g) b) : (⋃ x, Ioo (f x) (g x)) = Ioo a b :=
+  calc
+    (⋃ x, Ioo (f x) (g x)) = (⋃ x, Ioi (f x)) ∩ ⋃ x, Iio (g x) :=
+      unionᵢ_inter_of_monotone hf.Ioi hg.Iio
+    _ = Ioi a ∩ Iio b := congr_arg₂ (· ∩ ·) ha.unionᵢ_Ioi_eq hb.unionᵢ_Iio_eq
+#align Union_Ioo_of_mono_of_is_glb_of_is_lub unionᵢ_Ioo_of_mono_of_isGLB_of_isLUB
+
+end Union
+
```
Comment (dupuisf) [inline Mathlib/Data/Set/Intervals/Monotone.lean:199]:
```suggestion
end unionᵢ

```

**Review by dupuisf — APPROVED** [review]
Thanks!

bors d+

**Pull request closed without merging**